src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node top;
+    private Node bottom;
+
+    private void checkElementIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index doesn't exist:" + index);
+        }
+    }
+
+
+    private Node<T> getNode(int index) {
+        Node first = top;
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                return first;
+            }
+            first = first.next;
+        }
+        return null;
+    }
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-iterate-your-nodes-from-the-head-to-tail-or-vice-versa